### DISCUSSION TITLE
Xstate itegration

### DISCUSSION THREAD

**Discussion by bmadone** [pmndrs/jotai#1542] [Q&A] [answered]
Hi, I read documentation about xstate integration, but I don't get 1 thing, what if I configure my machine in a component, for example like this 

<img width="475" alt="CleanShot 2022-11-05 at 17 57 28@2x" src="https://user-images.githubusercontent.com/23120972/200128753-ba653701-631c-49b6-afa6-bcb5cf18dd32.png">

How can I create atom from it?

**Comment by dai-shi** [comment, marked as answer]
You can create an atom from it in render using `useMemo` or `useConstant`, but then it's not globally available.
Atoms are abstraction independent from React, so to make use of it, defining machine outside would fit better.
